test/modules/ModuleLib/ReporterClients/ReportEndpoint.cs

```diff
@@ -0,0 +1,10 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.ModuleUtil.ReporterClients
+{
+    public enum ReportEndpoint
```

Comment:
where to use it?  I think it is better to have a property "TestMode" in Settings; which will define what test this module is running, e.g. e2e, long haul, or connectivity test.  Then add more validation, e.g. ensure Test Result Coordinator Url is provided if test mode is connectivity test.